client/my-sites/posts/post-controls.jsx

```diff
@@ -167,6 +171,16 @@ module.exports = React.createClass( {
 			}
 		}
 
+		if ( post.status === 'publish' && utils.userCan( 'edit_post', post ) ) {
```

Comment:
I'm looking into it right now. I'll likely open a dedicated branch later, as it would basically mean rewriting the whole file from scratch.
Anyway, the idea is to move the logic outside `render` and to split it up into shorter and clearer methods (sort of like what it already happens in the editor drawer).